packages/client/test/client.spec.ts

```diff
@@ -91,11 +91,12 @@ const MOCK_PROVIDER: Provider = {
 };
 
 describe('OpenFeatureClient', () => {
-  beforeAll(() => {
+  beforeEach(() => {
```

Comment:
Is there a good reason for doing this each test?

Could we instead only do the `await OpenFeature.clearProviders();` in an afterAll?

Reply:
It's cleaner to reset the providers after each test, in my opinion. Also, some of the new tests change the provider and must be reset at some point. I could handle this another way if you prefer but I think a clean reset per test is worth the impact on testing performance.